src/main/java/seedu/address/logic/commands/LimitCommand.java

```diff
@@ -56,4 +56,19 @@ public CommandResult execute(Model model) throws CommandException {
         model.setThreshold(threshold);
         return new CommandResult(String.format(MESSAGE_LIMIT_THRESHOLD_SUCCESS, threshold));
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof LimitCommand)) {
+            return false;
+        }
+
+        LimitCommand otherLimitCommand = (LimitCommand) other;
+        return threshold.equals(otherLimitCommand.threshold);
+    }
```

Comment:
`equals` method added to `LimitCommand` in the functional code is **_STRICTLY ONLY USED FOR TESTING PURPOSES_** and is not used functionally in the program.